SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp

```diff
@@ -280,4 +280,54 @@ bool SpatialCommandUtils::HasDevLoginTag(const FString& DeploymentName, bool bIs
 	return false;
 }
 
+FProcHandle SpatialCommandUtils::StartLocalReceptionistProxyServer(bool bIsRunningInChina, const FString& CloudDeploymentName, const FString& ListeningAddress, const int32 Port , FString &OutResult, int32 &OutExitCode, bool &bProcessSucceeded)
+{
+	FString Command = FString::Printf(TEXT("cloud connect external %s --listening_address %s --local_receptionist_port %i"), *CloudDeploymentName, *ListeningAddress, Port);
+
+	if (bIsRunningInChina)
+	{
+		Command += SpatialGDKServicesConstants::ChinaEnvironmentArgument;
+	}
+
+	FProcHandle ProcHandle;
+
+	void* ReadPipe = nullptr;
+	void* WritePipe = nullptr;
+	ensure(FPlatformProcess::CreatePipe(ReadPipe, WritePipe));
+
+	ProcHandle = FPlatformProcess::CreateProc(*SpatialGDKServicesConstants::SpatialExe, *Command, false, true, true, nullptr, 1 /*PriorityModifer*/, *SpatialGDKServicesConstants::SpatialOSDirectory, WritePipe);
+
+	bProcessSucceeded = false;
+	bool bProcessFinished = false;
+	if (ProcHandle.IsValid())
+	{
+		while(!bProcessFinished && !bProcessSucceeded)
+		{
+			bProcessFinished = FPlatformProcess::GetProcReturnCode(ProcHandle, &OutExitCode);
+
+			OutResult = OutResult.Append(FPlatformProcess::ReadPipe(ReadPipe));
+			bProcessSucceeded = OutResult.Contains("The receptionist proxy is available");
+
+			FPlatformProcess::Sleep(0.01f);
+		}
+	}
+	else
+	{
+		UE_LOG(LogSpatialCommandUtils, Error, TEXT("%s"), *FText::Format(LOCTEXT( "CloudConnectExternalExecutionFailed","Execution failed. '{0}' with arguments '{1}' in directory '{2}' "), FText::FromString(SpatialGDKServicesConstants::SpatialExe), FText::FromString(Command), FText::FromString(SpatialGDKServicesConstants::SpatialOSDirectory)).ToString());
+	}
+
+	FPlatformProcess::ClosePipe(0, ReadPipe);
+	FPlatformProcess::ClosePipe(0, WritePipe);
+
+	return ProcHandle;
```

Comment:
I believe we should terminate the process immediately if it didn't succeed (I imagine it would terminate itself anyway but just to make sure) and return an empty handle to signify that. We would also not need to have the separate bool for success.